src/util/TypeIdentity.h

```diff
@@ -0,0 +1,13 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author:
+//   2024      Fabian Krause (fabian.krause@students.uni-freiburg.de)
+
+#pragma once
+
+#include <type_traits>
+
```

Comment:
How about two aliases:
One that is `TI` that is the type (so using TI = type_identity<T> without braces)
and then your `auto ti = TI<T>{}` (your current TI).
Then the following comment can be added (and adapt your usage).

Add comment:
```
The use case of the following aliases and constants is as follows:
Locally (inside a function) put a 
`using namespace ad_utility::use_type_identity`, 
then use `TI<someType>` as a convenient short cut for `std::type_identity<someType>{}`. This can be used to make template arguments to local lambdas more readable, as C++ doesn't allow for local templates, example.
auto f1 = []<typename T>() {/* use T */};
auto f2 = []<typename T>(TI<T>) { /* use T /*};
f1.template operator()<int>(); // Works but is rather unreadable.
f2(ti<int>); // Minimal visual overhead for specifying the template parameter.
```